src/android/CodePushReportingManager.java

```diff
@@ -44,26 +22,22 @@ public CodePushReportingManager(Activity cordovaActivity, CodePushPreferences co
     /**
      * Invokes the window.codePush.reportStatus JS function for the given webView.
      */
-    public void reportStatus(Status status, String label, String appVersion, String deploymentKey, final CordovaWebView webView) {
+    public void reportStatus(StatusReport statusReport, final CordovaWebView webView) {
         /* JS function to call: window.codePush.reportStatus(status: number, label: String, appVersion: String, currentDeploymentKey: String, previousLabelOrAppVersion?: string, previousDeploymentKey?: string) */
-        if (deploymentKey == null || deploymentKey.isEmpty()) {
+        if (statusReport.deploymentKey == null || statusReport.deploymentKey.isEmpty()) {
```

Comment:
oh this reminds me of Yahoo...we had a Util class full of a bunch of wrapper methods for situations like this...
```
public static void isEmptyOrNull(string a) {
    return a == null || a.isEmpty();
}
```
not sure how many times we do checks like this but it may be worth looking into adding some utility functions